web/components/layout/tabs.tsx

```diff
@@ -1,82 +1,121 @@
-                )}
-                aria-current={activeIndex === i ? 'page' : undefined}
-              >
-                <Row className={'items-center justify-center gap-1'}>
-                  {tab.tabIcon && <span> {tab.tabIcon}</span>}
-                  {tab.badge ? (
-                    <div className="px-0.5 font-bold">{tab.badge}</div>
-                  ) : null}
-                  {tab.title}
-                </Row>
-              </a>
-            </Link>
-          ))}
-        </nav>
-      </div>
-
+      <nav
+        className={clsx('mb-4 space-x-8 border-b border-gray-200', className)}
+        aria-label="Tabs"
+      >
+        {tabs.map((tab, i) => (
+          <a
+            href="#"
+            key={tab.title}
+            onClick={(e) => {
+              e.preventDefault()
+              track('Clicked Tab', {
+                title: tab.title,
+                currentPage: currentPageForAnalytics,
+              })
+              onClick?.(tab.title, i)
+            }}
+            className={clsx(
+              activeIndex === i
+                ? 'border-indigo-500 text-indigo-600'
+                : 'border-transparent text-gray-500 hover:border-gray-300 hover:text-gray-700',
+              'inline-flex cursor-pointer flex-row gap-1 whitespace-nowrap border-b-2 px-1 py-3 text-sm font-medium',
+              labelClassName
+            )}
+            aria-current={activeIndex === i ? 'page' : undefined}
+          >
+            {tab.tabIcon && <span>{tab.tabIcon}</span>}
+            {tab.badge ? (
+              <span className="px-0.5 font-bold">{tab.badge}</span>
+            ) : null}
+            {tab.title}
+          </a>
+        ))}
+      </nav>
       {activeTab?.content}
     </>
   )
 }
+
+export function ControlledTabs(props: TabProps & { defaultIndex?: number }) {
+  const { defaultIndex, onClick, ...rest } = props
+  const [activeIndex, setActiveIndex] = useState(defaultIndex ?? 0)
+  return (
+    <UncontrolledTabs
+      {...rest}
+      activeIndex={activeIndex}
+      onClick={(title, i) => {
+        setActiveIndex(i)
+        onClick?.(title, i)
+      }}
+    />
+  )
+}
+
+const isTabSelected = (router: NextRouter, queryParam: string, tab: Tab) => {
+  const selected = router.query[queryParam]
+  if (typeof selected === 'string') {
+    return tab.title.toLowerCase() === selected
+  } else {
+    return false
+  }
+}
+
+export function QueryControlledTabs(
+  props: TabProps & { defaultIndex?: number }
+) {
+  const { tabs, defaultIndex, onClick, ...rest } = props
+  const router = useRouter()
+  const selectedIdx = tabs.findIndex((t) => isTabSelected(router, 'tab', t))
+  const activeIndex = selectedIdx !== -1 ? selectedIdx : defaultIndex ?? 0
+  return (
+    <UncontrolledTabs
+      {...rest}
+      tabs={tabs}
+      activeIndex={activeIndex}
+      onClick={(title, i) => {
+        router.replace(
+          { query: { ...router.query, tab: title.toLowerCase() } },
```

Comment:
A minor point might be that if the default tab is later selected, it should not add the query string

Reply:
Another slight change that is up to personal taste would be using `?tab='bets' vs `?bets`.

I slightly prefer the latter b/c it's shorter, but I am ok with how it works now.

Reply:
I personally prefer the way it works now on these two points. If you feel strongly you can go change it later.